### ISSUE TITLE
[onert] Loss value difference after kernel replacement

### ISSUE THREAD

**Issue by ragmani** [Samsung/ONE#11341]
After applying the new kernel, loss values are better reduced. Why?

- after
```bash
$ ./Product/x86_64-linux.release/out/bin/onert_train mnist_keras.circle --load_expected:raw out/train.output.1000.bin --load_input:raw out/train.input.1000.bin --data_length 1000
Model Expected Filename out/train.output.1000.bin
Model Input Filename out/train.input.1000.bin
Model Filename mnist_keras.circle
[Epoch 0] Output [0] Loss: 84.2275
[Epoch 1] Output [0] Loss: 83.2126
[Epoch 2] Output [0] Loss: 82.2672
[Epoch 3] Output [0] Loss: 81.3875
[Epoch 4] Output [0] Loss: 80.5693
===================================
MODEL_LOAD   takes 0.445 ms
PREPARE      takes 2.685 ms
EXECUTE      takes 652.513 ms
- MEAN     :  652.513 ms
- MAX      :  652.513 ms
- MIN      :  652.513 ms
- GEOMEAN  :  652.513 ms
===================================
```

- before
```bash
$ ./Product/x86_64-linux.release/out/bin/onert_train mnist_keras.circle --load_expected:raw out/train.output.1000.bin --load_input:raw out/train.input.1000.bin --data_length 1000
Model Expected Filename out/train.output.1000.bin
Model Input Filename out/train.input.1000.bin
Model Filename mnist_keras.circle
[Epoch 0] Output [0] Loss: 84.5231
[Epoch 1] Output [0] Loss: 84.1669
[Epoch 2] Output [0] Loss: 83.8783
[Epoch 3] Output [0] Loss: 83.6532
[Epoch 4] Output [0] Loss: 83.4876
===================================
MODEL_LOAD   takes 0.569 ms
PREPARE      takes 2.628 ms
EXECUTE      takes 1090.428 ms
- MEAN     :  1090.428 ms
- MAX      :  1090.428 ms
- MIN      :  1090.428 ms
- GEOMEAN  :  1090.428 ms
===================================
```

_Originally posted by @ragmani in https://github.com/Samsung/ONE/issues/11270#issuecomment-1691045025_

The loss values decrease more after replacing the fc kernel with the new kernel, it seems that training is better.
We need to find out why there is the difference.

**Comment by ragmani** [comment]
I found out a bug using wrong shapes in `FullyConnectedLayer` for training. After applying a patch, the loss values are improved and the loss values of the new kernel are equivalent with the existing kernel.

```bash
$ ./Product/x86_64-linux.release/out/bin/onert_train mnist_keras.circle --load_expected:raw out/train.output.1000.bin --load_input:raw out/train.input.1000.bin --data_length 1000
Model Expected Filename out/train.output.1000.bin
Model Input Filename out/train.input.1000.bin
Model Filename mnist_keras.circle
[Epoch 0] Output [0] Loss: 65.2821
[Epoch 1] Output [0] Loss: 41.2443
[Epoch 2] Output [0] Loss: 30.8735
[Epoch 3] Output [0] Loss: 25.4364
[Epoch 4] Output [0] Loss: 22.1303
===================================
MODEL_LOAD   takes 0.712 ms
PREPARE      takes 2.764 ms
EXECUTE      takes 636.741 ms
- MEAN     :  636.741 ms
- MAX      :  636.741 ms
- MIN      :  636.741 ms
- GEOMEAN  :  636.741 ms
===================================
```

```bash
$ ./Product/x86_64-linux.release/out/bin/onert_train mnist_keras.circle --load_expected:raw out/train.output.1000.bin --load_input:raw out/train.input.1000.bin --data_length 1000
Model Expected Filename out/train.output.1000.bin
Model Input Filename out/train.input.1000.bin
Model Filename mnist_keras.circle
[Epoch 0] Output [0] Loss: 65.2821
[Epoch 1] Output [0] Loss: 41.2443
[Epoch 2] Output [0] Loss: 30.8735
[Epoch 3] Output [0] Loss: 25.4364
[Epoch 4] Output [0] Loss: 22.1303
===================================
MODEL_LOAD   takes 0.450 ms
PREPARE      takes 2.827 ms
EXECUTE      takes 1342.684 ms
- MEAN     :  1342.684 ms
- MAX      :  1342.684 ms
- MIN      :  1342.684 ms
- GEOMEAN  :  1342.684 ms
===================================
```

**Comment by ragmani** [comment]
Resolved.
